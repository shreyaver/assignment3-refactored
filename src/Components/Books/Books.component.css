
.books {
    flex: 10;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 0;
}
.booksContent {
    flex: 1;
    padding: 0;
    width: 70%;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
}