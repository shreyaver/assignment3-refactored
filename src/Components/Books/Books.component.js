import React from 'react';
import PropTypes from 'prop-types';
import BooksByAuthor from '../BooksByAuthor/BooksByAuthor.component';
import './Books.component.css';

const Books = (props) => {
  const authors = [];
  const { books, handleLikeClick } = props;
  Object.keys(books).forEach((author) => {
    authors.push(<BooksByAuthor author={author} books={books[author]} handleLikeClick={handleLikeClick} key={author} />);
  });
  return (
    <div className="books">
      <div className="booksContent">
        {authors}
      </div>
    </div>
  );
};
Books.propTypes = {
  books: PropTypes.objectOf(
    PropTypes.arrayOf(
      PropTypes.shape({
        Name: PropTypes.string.isRequired,
        rating: PropTypes.number.isRequired,
        Author: PropTypes.string.isRequired,
        id: PropTypes.number.isRequired,
      }).isRequired,
    ).isRequired,
  ).isRequired,
  handleLikeClick: PropTypes.func.isRequired,
};
export default Books;
