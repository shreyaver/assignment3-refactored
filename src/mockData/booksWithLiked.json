{ "books": {
  "J K Rowling":
    [{ "Author": "J K Rowling",
        "id":70,
        "Name":"Harry Potter and the Deathly Hallows (Harry Potter, #7)",
        "rating":4.62,
        "liked": false
     },
     {"Author": "J K Rowling",
      "id":30,
      "Name":"Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)",
      "rating":4.54,
      "liked": false
     },
     {"Author":"J K Rowling",
      "id":60,
      "Name":"Harry Potter and the Half-Blood Prince (Harry Potter, #6)",
      "rating":4.54,
      "liked": false
    }],
  "Sidney Sheldon":
    [{"Author":"Sidney Sheldon",
      "id":90,
      "Name":"Master of the Game",
      "rating":4.1,
      "liked": false
     },
     {"Author":"Sidney Sheldon",
      "id":80,
      "Name":"If Tomorrow Comes (Tracy Whitney Series, #1)",
      "rating":4.02,
      "liked": false
     },
     {"Author":"Sidney Sheldon",
      "id":100,
      "Name":"Tell Me Your Dreams",
      "rating":3.93,
      "liked": false
     }]
}}