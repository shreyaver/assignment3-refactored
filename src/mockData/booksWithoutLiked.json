{ "J K Rowling":
    [{ "Author": "J K Rowling",
        "id":70,
        "Name":"Harry Potter and the Deathly Hallows (Harry Potter, #7)",
        "rating":4.62
     },
     {"Author": "J K Rowling",
      "id":30,
      "Name":"Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)",
      "rating":4.54
     },
     {"Author":"J K Rowling",
      "id":60,
      "Name":"Harry Potter and the Half-Blood Prince (Harry Potter, #6)",
      "rating":4.54
    }],
  "Sidney Sheldon":
    [{"Author":"Sidney Sheldon",
      "id":90,
      "Name":"Master of the Game",
      "rating":4.1
     },
     {"Author":"Sidney Sheldon",
      "id":80,
      "Name":"If Tomorrow Comes (Tracy Whitney Series, #1)",
      "rating":4.02
     },
     {"Author":"Sidney Sheldon",
      "id":100,
      "Name":"Tell Me Your Dreams",
      "rating":3.93
     }]
}