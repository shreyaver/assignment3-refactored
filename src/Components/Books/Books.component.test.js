import React from 'react';
import renderer from 'react-test-renderer';
import Books from './Books.component';
import mockData from '../../mockData/booksWithLiked.json';

describe('Books', () => {
  it('renders without crashing', () => {
    const books = renderer.create(<Books books={mockData.books} handleLikeClick={jest.fn()} />).toJSON();
    expect(books).toMatchSnapshot();
  });
});
