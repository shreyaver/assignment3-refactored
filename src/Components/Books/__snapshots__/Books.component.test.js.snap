// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books renders without crashing 1`] = `
<div
  className="books"
>
  <div
    className="booksContent"
  >
    <div
      className="booksByAuthorContainer"
    >
      <div
        className="authorName"
      >
        <span>
          J K Rowling
        </span>
      </div>
      <div
        className="booksByAuthor"
      >
        <div
          className="book"
          style={
            Object {
              "backgroundImage": "url(cover_image.jpeg)",
            }
          }
        >
          <div
            className="imagePanel"
          />
          <div
            className="bookData"
          >
            <p
              className="bookName"
            >
              Harry Potter and the Deathly Hallows (Harry Potter, #7)
            </p>
            <span>
              4.6
              <button
                className="heartIcon"
                type="button"
              />
            </span>
          </div>
        </div>
        <div
          className="book"
          style={
            Object {
              "backgroundImage": "url(cover_image.jpeg)",
            }
          }
        >
          <div
            className="imagePanel"
          />
          <div
            className="bookData"
          >
            <p
              className="bookName"
            >
              Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)
            </p>
            <span>
              4.5
              <button
                className="heartIcon"
                type="button"
              />
            </span>
          </div>
        </div>
        <div
          className="book"
          style={
            Object {
              "backgroundImage": "url(cover_image.jpeg)",
            }
          }
        >
          <div
            className="imagePanel"
          />
          <div
            className="bookData"
          >
            <p
              className="bookName"
            >
              Harry Potter and the Half-Blood Prince (Harry Potter, #6)
            </p>
            <span>
              4.5
              <button
                className="heartIcon"
                type="button"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="booksByAuthorContainer"
    >
      <div
        className="authorName"
      >
        <span>
          Sidney Sheldon
        </span>
      </div>
      <div
        className="booksByAuthor"
      >
        <div
          className="book"
          style={
            Object {
              "backgroundImage": "url(cover_image.jpeg)",
            }
          }
        >
          <div
            className="imagePanel"
          />
          <div
            className="bookData"
          >
            <p
              className="bookName"
            >
              Master of the Game
            </p>
            <span>
              4.1
              <button
                className="heartIcon"
                type="button"
              />
            </span>
          </div>
        </div>
        <div
          className="book"
          style={
            Object {
              "backgroundImage": "url(cover_image.jpeg)",
            }
          }
        >
          <div
            className="imagePanel"
          />
          <div
            className="bookData"
          >
            <p
              className="bookName"
            >
              If Tomorrow Comes (Tracy Whitney Series, #1)
            </p>
            <span>
              4
              <button
                className="heartIcon"
                type="button"
              />
            </span>
          </div>
        </div>
        <div
          className="book"
          style={
            Object {
              "backgroundImage": "url(cover_image.jpeg)",
            }
          }
        >
          <div
            className="imagePanel"
          />
          <div
            className="bookData"
          >
            <p
              className="bookName"
            >
              Tell Me Your Dreams
            </p>
            <span>
              3.9
              <button
                className="heartIcon"
                type="button"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
